src/csharp/Intel/Generator/Assembler/CSharp/CSharpAssemblerSyntaxGenerator.cs

```diff
@@ -0,0 +1,702 @@
+						writer.WriteLine($"if (Bitness {(has64 ? "==" : ">=")} {bitness}) {{");
+
+						using (writer.Indent()) {
+							writer.WriteLine($"op = Code.{legacy.Code.Name(Converter)};");
+						}
+
+						writer.Write("}");
+						needToThrowIfNotFound = true;
+					}
+					else {
+						writer.WriteLine($"op = Code.{opCodeInfo.Code.Name(Converter)};");
+					}
+				}
+			}
+
+			if (needToThrowIfNotFound) {
+				writer.WriteLine(" else {");
+				using (writer.Indent()) {
+					writer.Write($"throw NoOpCodeFoundFor(Mnemonic.{group.MemoName}");
+					for (var i = 0; i < args.Count; i++) {
+						var renderArg = args[i];
+						if (renderArg.Kind == ArgKind.HiddenMemory) continue;
+						writer.Write(", ");
+						writer.Write(renderArg.Name);
+					}
+					writer.WriteLine(");");
+				}
+				writer.WriteLine("}");
+			}
+		}
+
+		static string GetArgConditionForOpCode(OpCodeInfo opCode) {
+			if (opCode is VexOpCodeInfo) {
+				return $"PreferVex";
+			}
+			else if (opCode is EvexOpCodeInfo) {
+				return $"!PreferVex";
+			}
+			return $"unsupported_condition_for_op_code_{opCode.Code.RawName}";
+		}
+
+		static string GetArgConditionForOpCodeKind(RenderArg arg, CommonOpKind opKind) {
+			var regName = arg.Name;
+			var isMemory = arg.Kind == ArgKind.RegisterMemory || arg.Kind == ArgKind.Memory;
+			switch (opKind) {
+			case CommonOpKind.CL:
+			case CommonOpKind.AL:
+			case CommonOpKind.AX:
+			case CommonOpKind.EAX:
+			case CommonOpKind.RAX:
+			case CommonOpKind.ES:
+			case CommonOpKind.CS:
+			case CommonOpKind.SS:
+			case CommonOpKind.DS:
+			case CommonOpKind.FS:
+			case CommonOpKind.GS:
+			case CommonOpKind.DX:
+				return $"{regName} == Register.{opKind}";
+			case CommonOpKind.r8_rb:
+			case CommonOpKind.Gb:
+			case CommonOpKind.Eb:
+			case CommonOpKind.RdMb:
+			case CommonOpKind.RqMb:
+			case CommonOpKind.Mb:
+				return isMemory ? $"{regName}.Size == MemoryOperandSize.BytePtr" : $"{regName}.IsGPR8()";
+			case CommonOpKind.r16_rw:
+			case CommonOpKind.Gw:
+			case CommonOpKind.Ew:
+			case CommonOpKind.RdMw:
+			case CommonOpKind.RqMw:
+			case CommonOpKind.Mw:
+				return isMemory ? $"{regName}.Size == MemoryOperandSize.WordPtr" : $"{regName}.IsGPR16()";
+			case CommonOpKind.r32_rd:
+			case CommonOpKind.Hd:
+			case CommonOpKind.Gd:
+			case CommonOpKind.Ed:
+			case CommonOpKind.Rd:
+			case CommonOpKind.Md:
+				return isMemory ? $"{regName}.Size == MemoryOperandSize.DwordPtr" : $"{regName}.IsGPR32()";
+			case CommonOpKind.r64_ro:
+			case CommonOpKind.Hq:
+			case CommonOpKind.Gq:
+			case CommonOpKind.Eq:
+			case CommonOpKind.Rq:
+			case CommonOpKind.Mq:
+				return isMemory ? $"{regName}.Size == MemoryOperandSize.QwordPtr" : $"{regName}.IsGPR64()";
+			case CommonOpKind.Imm1:
+				return $"{regName} == 1";			
+			case CommonOpKind.Ib:
+			case CommonOpKind.Ib16:
+			case CommonOpKind.Ib32:
+			case CommonOpKind.Ib64:
+				return $"{regName} >= sbyte.MinValue && {regName} <= byte.MaxValue";			
```

Comment:
If it's `Ib`, this is correct, but Ib16/32/64 sign extend the 8-bit immediate. Eg. if it's Iw and the byte is 0xFF it will be sign extended to 0xFFFF so if it's Ib16/32/64, the check should be >= sbyte.Min and <= sbyte.Max.

Reply:
Oh, ok, right, so I will have to disambiguate sign extended up-front as well to detect them for the group. This part of the ode is actually no longer used as it is done in one shot as a group now.